### ISSUE TITLE
MMM-Todoist stuck at loading...

### ISSUE THREAD

**Issue by martyn-holland** [cbrooker/MMM-Todoist#93]
Pulled the git and ran npm install. Get these two warning:

- npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
- npm WARN deprecated har-validator@5.1.5: this library is no longer supported

_Dont know if this is the problem_

Tried adding the module and have tried resetting the API key but no joy.

Heres config file:

`        {
                 module: 'MMM-Todoist',
                 position: 'top_left',   // This can be any of the regions. Best results in left or right regions.
                 header: 'Todo List', // This is optional
                 config: { // See 'Configuration options' for more information.
                         hideWhenEmpty: false,
                         accessToken: 'apikeyremoved',
                         maximumEntries: 60,
                         updateInterval: 10*60*1000, // Update every 10 minutes
                         fade: true,
                         displayLastUpdate: true,
                         displayAvatar: true,
                         // projects and/or labels is mandatory:
                         //projects: [ 166564794 ],
                         labels: [ "MagicMirror", "Important" ]
       }
  }`

**Comment by mike-link** [comment]
Yep it looks like the there was something that broke it for me too. We have 3 lists and one of them has categories and that is the only one the broke. Hopefully that points people in the right direction. I ended up reverting back to the commit that we know worked. we did this by running
`git checkout df2d4ff` 

Hopefully there is a fix for this soon.

**Comment by robertybob** [comment]
I am having this same issue, having installed the module for the first time. Forgive me @mike-link , but where do I run that git checkout code from? I tried it within the module's folder but I was presented with some error message

**Comment by mike-link** [comment]
In the directory that you installed/clone the module in. So for me it is `<magic-mirror-root>/modules/mmm-todoist`

**Comment by robertybob** [comment]
@mike-link  I get this error message when I try that, do you know what it means?

![image](https://user-images.githubusercontent.com/2610395/123070494-b7e68a00-d40b-11eb-9c7b-94176bfed019.png)

**Comment by sdetweil** [comment]
@mike-link 
the module name MUST match the folder name and the register name in the modulename.js
yours does: all three are named MMM-ToDoIst, exact case matters... and git clone will build a folder that matches the repo name (by default)

if there was a name problem, you wouldn't even get loading...

**Comment by kevinleberge** [comment]
I'm having the same issue.  Stuck at loading...

I ran the checkout command and also got that same error:
```
git checkout df2d4ff
error: Your local changes to the following files would be overwritten by checkout:
	package-lock.json
	package.json
Please commit your changes or stash them before you switch branches.
Aborting
```


I microsoft todo, and todoist. Neither of them work both stuck at loading...   man you'd think getting a todo list into magic mirror would be easier.. I'm exhausted spent hours on this already :(

**Comment by sdetweil** [comment]
I just installed todoist module, got the access token from the app  in the todoist page from  the readme link, added a project, got its number, added it to the projects list, 
added a task in that project and refreshed MM and boom its there..

never used this module before

**Comment by sdetweil** [comment]
and just added the skill to alexa, added a task and then go the alexa todolist project id, added it to the todoist config and voila (middle right) 
![Screenshot at 2021-08-06 23-06-36](https://user-images.githubusercontent.com/1498142/128587382-41826dff-129a-4c28-8560-8e9a938429a4.png)

**Comment by cbrooker** [comment]
> I'm having the same issue. Stuck at loading...
> 
> I ran the checkout command and also got that same error:
> 
> ```
> git checkout df2d4ff
> error: Your local changes to the following files would be overwritten by checkout:
> 	package-lock.json
> 	package.json
> Please commit your changes or stash them before you switch branches.
> Aborting
> ```
> 
> I tried microsoft todo, and todoist. Neither of them work both stuck at loading... man you'd think getting a todo list into magic mirror would be easier.. I'm exhausted spent hours on this already :(

Sounds like a git conflict. Not sure what changed your local package file.
Go to the MMM-Todist folder and type "git stash". 
Then do your checkout again. "git checkout df2d4ff"

Looking at the thread. I see there is a problem with the latest build. I'll try to take a look soon.

Chris

**Comment by sdetweil** [comment]
@cbrooker 
I just installed for the very 1st time, and it worked right away..  on MM 2.16

**Comment by Nickred896** [comment]
I had the loading problem when I first installed. Ran through all troubleshooting in this thread. I got this module to work only after deleting the MMM-Todoist folder from modules and going back through the install steps.

**Comment by RickSanchezC136** [comment]
Having the same issue:

Created new project in todoist.
Copied API key and project number and pasted into module config.
ran Git Clone command to pull module, then went into folder and ran npm install.
Just says loading...

I used this module over a year ago and it was working. I'm not sure why it's not working now.
I dont have any other modules installed. Fresh install (last week). Really a bummer, I can't get google tasks working.

**Comment by sdetweil** [comment]
I'll look at.mine in an hour or so. I don't remember anything special about either task or todoist..

Google u have to add a credit card to make any API work

**Comment by sdetweil** [comment]
I just installed MMM-Todoist, on latest MMM 2.20.0

copied api key from web site (called accesstoken in config) 
got project numbers from web site , pasted numbers (no quotes) 

worked on 1st try 
(accessToken mangled here) 
```
		{
		module: 'MMM-Todoist',
		position: 'top_right',	// This can be any of the regions. Best results in left or right regions.
		header: 'Todoist', // This is optional
		config: { // See 'Configuration options' for more information.
			hideWhenEmpty: false,
			accessToken: '4e27xx2345e124bcdfda0dad05a7ad2fcb1c24',
			maximumEntries: 60,
			updateInterval: 10*60*1000, // Update every 10 minutes
			fade: false,
			// projects and/or labels is mandatory:
			projects: [ 2270986771 , 2209919893],
			labels: [ "MagicMirror", "Important" ] // Tasks for any projects with these labels will be shown.
      }
	}
```

**Comment by ACiD2142** [comment]
Hi

I installed this module today suck on loading can any one please assist

Thank you

**Comment by pauabaer** [comment]
Hi guys..i have the same problem. is there any solution right now?

**Comment by Jdwilcox211** [comment]
I have same problem just happe3d after most recent pull.  Tried to change api key and I delete and recloned the module to no success I even tried to check out previous commit but it does seem to work

**Comment by karldonteljames** [comment]
Another first-time installer here having the same issue. 
I've removed the MMM-Todoist folder and re-cloned it, but no change. Is there a log file anywhere?
I've also double checked the config, and tried to swap between projects and labels but still sits at loading. - Config below.
```
{
      "module": "MMM-Todoist",
      "header": "Dads Todo List",
      "position": "top_right",
      "config": {
        "hideWhenEmpty": false,
        "accessToken": "MySuperSecretAccessToken",
        "maximumEntries": 20,
        "updateInterval": 300000,
        "fade": false,
        "labels": [
          "MagicMirror",
          "Important"
        ],
        "displayLastUpdate": true,
        "maxtitleLength": 20
      }
    }
    ```

**Comment by karldonteljames** [comment]
I've just seent that the API is now on version 9 (another open issue) and i've added  
```
apiversion: 'v9',
apiBase: "https://todoist.com/API"
```
to the config, but this also makes no difference.

**Comment by karldonteljames** [comment]
Hey all, for those still having this issue, i think it may be related to the v9 API. I've just changed my config and tripple checked my access token, projects, and tags, and this is now working.

```
                {
                        module: 'MMM-Todoist',
                        position: 'top_right',
                        header: "To-Do List",
                        config: {
                                hideWhenEmpty: false,
                                accessToken: "MyPrivateAccessTokenWithQuotes",
                                apiVersion: "v9",
                                apiBase: "https://todoist.com/API",
                                todoistEndpoint: "sync",
                                maximumEntries: 60,
                                updateInterval: 10*60*1000,
                                fade: true,
                                labels:  [ "MagicMirror", "Important" ],
                                projects: [ "2304370707" ],
                                sortType: "dueDateDescPriority",
                                maxTitleLength: 1000,
                                wrapEvents: false,
                                showProject: false
                        }
                }
```

**Comment by karldonteljames** [comment]
I know the last few posts have been mine, but i can confirm that you do not need both labels and projects.
I'm pulling from projects, and have three seperate lists on on my MagicMirror now. Working brilliantly.

**Comment by waterglass11** [comment]
Hi all, I've tried the "git stash" and "git checkout" here and although I did received a package-lock.json notification when I first tried the checkout, I'm not seeing it after the stash command followed by the checkout.

The problem is I've copied @karldonteljames module text verbatim into my config, changed the project name and token to be my own info, and I'm still stuck at "Loading..." instead of pulling up tasks. Anyone else have any ideas what could be causing the issue?

**Comment by waterglass11** [comment]
Just kidding - see my reply [here](https://github.com/cbrooker/MMM-Todoist/issues/103#issuecomment-1482147250). I rebuilt the mirror from scratch and now it works fine!
